src/rustdoc_generation.rs

```diff
@@ -7,44 +7,50 @@ use crate::dump::RustDocCommand;
 use crate::util::slugify;
 use crate::GlobalConfig;
 
-pub(crate) trait BaselineLoader {
-    fn load_rustdoc(
+pub(crate) trait RustdocGenerator {
+    fn generate_rustdoc(
         &self,
         config: &mut GlobalConfig,
-        rustdoc: &RustDocCommand,
+        rustdoc_command: &RustDocCommand,
         name: &str,
-        version_current: Option<&semver::Version>,
+        highest_considered_registry_version: Option<&semver::Version>,
+        crate_type: &str,
     ) -> anyhow::Result<std::path::PathBuf>;
 }
 
-pub(crate) struct RustdocBaseline {
+pub(crate) struct FromRustdoc {
```

Comment:
I'm overall not a fan of the `From` prefixes in all these names. While `rustdoc_generation::FromRustdoc` etc. seems pretty obvious, there's zero guarantee that it will always be used like that and it seems brittle from an understandability perspective to assume it. For example, `Debug` would only print the name of the type itself.

Consider the `RustdocFrom*` prefix? Then this struct could be `RustdocFromFile` etc.

On a separate note, I'm astonished to see that these structs aren't `#[derive(Debug)]`. Please add that throughout?